docs/reference/glossary.rst

```diff
@@ -4,152 +4,190 @@ Glossary
-        i.e. becoming slower and slower over time) and that the space
-        used for interned strings is not reclaimable. String interning
-        is done automatically by MicroPython compiler and runtimer when
-        it's either required by the implementation (e.g. function keyword
-        arguments are represented by interned string id's) or deemed
-        beneficial (e.g. for short enough strings, which have a chance
-        to be repeated, and thus interning them would save memory on
-        copies). Most of string and I/O operations don't produce interned
-        strings due to drawbacks described above.
+        An optimisation used by MicroPython to improve the efficiency of
+        working with strings. An interned string is referenced by its (unique)
+        identity rather than its address and can therefore be quickly compared
+        just by its identifier. It also means that identical strings can be
+        de-duplicated in memory. String interning is almost always invisible to
+        the user.
 
     MCU
         Microcontroller. Microcontrollers usually have much less resources
-        than a full-fledged computing system, but smaller, cheaper and
+        than a desktop, laptop, or phone, but are smaller, cheaper and
         require much less power. MicroPython is designed to be small and
         optimized enough to run on an average modern microcontroller.
 
     micropython-lib
         MicroPython is (usually) distributed as a single executable/binary
         file with just few builtin modules. There is no extensive standard
-        library comparable with :term:`CPython`. Instead, there is a related, but
-        separate project
-        `micropython-lib <https://github.com/micropython/micropython-lib>`_
-        which provides implementations for many modules from CPython's
-        standard library. However, large subset of these modules require
-        POSIX-like environment (Linux, FreeBSD, MacOS, etc.; Windows may be
-        partially supported), and thus would work or make sense only with
-        `MicroPython Unix port`. Some subset of modules is however usable
-        for `baremetal` ports too.
-
-        Unlike monolithic :term:`CPython` stdlib, micropython-lib modules
-        are intended to be installed individually - either using manual
-        copying or using :term:`upip`.
+        library comparable with :term:`CPython`'s. Instead, there is a related,
+        but separate project `micropython-lib
+        <https://github.com/micropython/micropython-lib>`_ which provides
+        implementations for many modules from CPython's standard library.
+        However, a large subset of these modules require POSIX-like
+        environment (Linux, FreeBSD, MacOS, etc.; Windows may be partially
+        supported), and thus would work or make sense only on the `MicroPython
+        Unix port`. Some subset of modules are usable for the
```

Comment:
Maybe add a short explanation that the main reason for this (i.e. most of it only being usable on POSIX-like) is that's it is implemented by ffi?